src/cdp/webSocketTransport.ts

```diff
@@ -30,18 +30,34 @@ export class WebSocketTransport implements ITransport {
 
     while (true) {
       try {
-        const ws = new WebSocket(url, [], {
+        const options = {
           headers: { host: 'localhost' },
           perMessageDeflate: false,
           maxPayload: 256 * 1024 * 1024, // 256Mb
           rejectUnauthorized: !(isSecure && targetAddressIsLoopback),
           followRedirects: true,
-        });
+        };
 
+        const ws = new WebSocket(url, [], options);
         return await timeoutPromise(
           new Promise<WebSocketTransport>((resolve, reject) => {
             ws.addEventListener('open', () => resolve(new WebSocketTransport(ws)));
-            ws.addEventListener('error', errorEvent => reject(errorEvent.error)); // Parameter is an ErrorEvent. See https://github.com/websockets/ws/blob/master/doc/ws.md#websocketonerror
+            ws.addEventListener('error', errorEvent => {
+              // Check for invalid http redirects for compatibility with old cdp proxies
+              const redirectedUrl = ws.url.replace(/^http:/, 'ws:').replace(/^https:/, 'wss:');
+
+              if (redirectedUrl === url) {
+                reject(errorEvent); // Parameter is an ErrorEvent. See https://github.com/websockets/ws/blob/master/doc/ws.md#websocketonerror
+                return;
+              }
+
+              // Try to reconnect with the redirected URL
+              const redirectedSocket = new WebSocket(redirectedUrl, [], options);
```

Comment:
I suggest instead making a recursive call to the `create` method here so that timeout/cancellation is handled